adapter/src/main/java/com/twilio/kudu/sql/KuduEnumerable.java

```diff
@@ -511,16 +511,16 @@ public Schema getTableSchema() {
         tokenBuilder.addPredicate(predicate.toPredicate(calciteKuduTable));
       });
       return tokenBuilder.build();
-    }).flatMap(tokens -> {
-      return tokens.stream().map(token -> {
-        try {
-          return KuduScannerUtil.deserializeIntoAsyncScanner(token.serialize(), client,
-              calciteKuduTable.getKuduTable());
-        } catch (java.io.IOException ioe) {
-          throw new RuntimeException("Failed to setup scanner from token.", ioe);
-        }
-      });
-    }).collect(Collectors.toList());
+    }).flatMap(tokens -> tokens.stream().map(token -> {
+      try {
+        KuduScanner scanner = token.intoScanner(client.syncClient());
+        Field asyncScannerField = KuduScanner.class.getDeclaredField("asyncScanner");
```

Comment:
While I have always disliked use of that ScannerUtil class and hoped that code would some day live in Kudu itself, but this code makes our code flow go from async to sync and back to async(pseudo async).
Hence, I feel we still need to keep that old class around as this may manifest at load by making all the scanners get realized.

Reply:
Can you elaborate on what you mean by "manifest at load by making all the scanners get realized"? KuduScanner is just a wrapper around AsyncKuduScanner that calls `joinAndHandleException` on the non blocking methods.

Reply:
Cool.. I misunderstood some parts of the Scanner and this looks good now.
Unless we call `hasMoreRows` or `nextRows`, the blocking parts are not exposed.